identity-appsupport/src/commonMain/kotlin/com/android/identity/appsupport/ui/x509chain/X509VewData.kt

```diff
@@ -0,0 +1,127 @@
+package com.android.identity.appsupport.ui.x509chain
+
+import com.android.identity.asn1.ASN1Integer
+import com.android.identity.asn1.OID
+import com.android.identity.crypto.Crypto
+import com.android.identity.crypto.EcCurve
+import com.android.identity.crypto.X500Name
+import com.android.identity.crypto.X509Cert
+import com.android.identity.util.unsignedBigIntToString
+import com.android.identity.util.toHex
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.format
+import kotlinx.datetime.format.DateTimeComponents
+import kotlinx.datetime.offsetAt
+import kotlin.time.Duration.Companion.hours
+
+/** All data fields of the Cert as needed for display as text strings. */
+data class X509VewData(
```

Comment:
With `X509CertChainViewer` being renamed to `CertificateViewer` this needs a better name, probably `CertificateViewerData`?